### PR TITLE
Use invoke in Opcodes.execute

### PR DESCRIPTION
**Pull request by ClementWalter**

### CODE REVIEW

File: src/onlydust/imhotep/opcodes/add.cairo [hunk 1]
```diff
@@ -2,11 +2,11 @@
 
 from src.onlydust.imhotep.stack import StackStruct, Stack
 from starkware.cairo.common.uint256 import uint256_add
-from starkware.cairo.common.bool import TRUE
+
 func ADD{range_check_ptr, stack: StackStruct}() {
     alloc_locals;
-    let (local a) = Stack.pop();
     let (local b) = Stack.pop();
+    let (local a) = Stack.pop();
```
Comment (Bernardstanislas) [inline src/onlydust/imhotep/opcodes/add.cairo:9]: Why changing this?

File: src/onlydust/imhotep/opcodes/library.cairo [hunk 2]
```diff
@@ -163,19 +169,28 @@ namespace Opcodes {
...
-        }
-        if (key == MUL_CODE) {
-            MUL();
-            return ();
-        }
+    func build() -> (opcodes: codeoffset*) {
+        let (opcodes: codeoffset*) = alloc();
+        assert opcodes[0] = STOP;
+        assert opcodes[1] = ADD;
+        assert opcodes[2] = MUL;
```
Comment (Bernardstanislas) [inline src/onlydust/imhotep/opcodes/library.cairo:176]: Do you confirm we can set any index in `opcodes` even if it's not continuous?

**Review by Bernardstanislas — APPROVED** [review]

File: src/onlydust/imhotep/opcodes/add.cairo (hunk 1 above)
Comment (ClementWalter) [reply src/onlydust/imhotep/opcodes/add.cairo:9]: this is a mistake, I should have updated the test instead. The idea was to fix the fact that 

```
Stack.push(a)
Stack.push(b)
ADD()
```

is indead `ADD(b, a)` and not `ADD(a, b)`

File: src/onlydust/imhotep/opcodes/library.cairo (hunk 2 above)
Comment (ClementWalter) [reply src/onlydust/imhotep/opcodes/library.cairo:176]: I confirm event though it creates memory holes

**Comment by ClementWalter** [comment]
actually, I see no reason why this implementation is better atm, it creates more memory holes and does not really save lines of code because we still need to fill the array, which is somehow equivalent to write 256 `if .. return`

I'll leave it in a branch so that we can change mind later one should we discover something else

**Comment by ClementWalter** [comment]
Branch is here: https://github.com/ClementWalter/imhotep/tree/cw/opcodes/use-build-and-invoke

**Comment by Bernardstanislas** [comment]
🙏

**Pull request closed without merging**